### PR TITLE
Create LitTrait and remove repetitive code

### PR DESCRIPTION
**Pull request by DebHugo**
## Introduction
Create `LitTrait` for remove repetitive code

### API changes
Create `LitTrait` trait

### CODE REVIEW

**Comment by SOF3** [comment]
If even a getter/setter is repetitive and it is public anyway, why do we even have a getter/setter for this instead of exposing the field directly?

**Comment by DebHugo** [comment]
> If even a getter/setter is repetitive and it is public anyway, why do we even have a getter/setter for this instead of exposing the field directly?

Currently, the lit value is protected. Why is it not public? I don't know.
I am making this PR to continue the logic in pmmp. Indeed, there is `AnyFacingTrait` which has the same utility as `LitTrait` which I created.

File: src/block/RedstoneTorch.php [hunk 1]
```diff
@@ -23,28 +23,17 @@
 
 namespace pocketmine\block;
 
+use pocketmine\block\utils\LitTrait;
 use pocketmine\data\runtime\RuntimeDataDescriber;
 
 class RedstoneTorch extends Torch{
-	protected bool $lit = true;
+	use LitTrait;
```
Comment (dktapps) [inline src/block/RedstoneTorch.php]: This one is problematic. Its default state should be lit=true.

**Comment by dktapps** [comment]
> If even a getter/setter is repetitive and it is public anyway, why do we even have a getter/setter for this instead of exposing the field directly?

The setter is fluent, and the convention is to use functions for this for validation and such. There is an argument to be made for making the fields public directly (it would simplify implementation of the NBT blockstate serializers), but right now it's not practical and would make the API inconsistent.

**Review by ShockedPlot7560 — DISMISSED** [review]

File: src/block/utils/LitTrait.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block\utils;
+
+use pocketmine\data\runtime\RuntimeDataDescriber;
+
+trait LitTrait{
```
Comment (dktapps) [inline src/block/utils/LitTrait.php]: `LightableTrait` would be more in line with other similar traits.
Welcome to suggest a better name (I don't like my suggestion much) but I don't think `LitTrait` is quite accurate.

**Review by dktapps — APPROVED** [review]

**Review by ShockedPlot7560 — APPROVED** [review]

**Pull request merged**